snooty/parser.py

```diff
@@ -136,24 +136,65 @@ def __init__(
     def dispatch_visit(self, node: docutils.nodes.Node) -> None:
         line = util.get_line(node)
 
-        if isinstance(node, (docutils.nodes.definition, docutils.nodes.field_list)):
+        if isinstance(node, docutils.nodes.definition):
+            return
+        if isinstance(node, docutils.nodes.field_list):
+            top = self.state[-1]
+            if isinstance(top, n.Root):
+                for field in node.children:
+                    key = field.children[0].astext()
+                    value = field.children[1].astext()
+                    top.options[key] = value
+                raise docutils.nodes.SkipNode()
+
+            self.state.append(n.FieldList((line,), []))
             return
         elif isinstance(node, docutils.nodes.document):
             self.state.append(n.Root((0,), [], {}))
             return
         elif isinstance(node, docutils.nodes.field):
-            key = node.children[0].astext()
-            value = node.children[1].astext()
-            top = self.state[-1]
-            if isinstance(top, n.Root):
-                top.options[key] = value
-            else:
+            field_list = node.parent
+            assert isinstance(field_list, docutils.nodes.field_list)
+            rstobject = field_list.parent
+            field_name = node.children[0].astext()
+            try:
+                assert isinstance(rstobject, rstparser.directive)
+
+                # Convert list of mixed strings and tuples into a key: value map
+                supported_fields = {
+                    (f if isinstance(f, str) else f[0]): (
+                        None if isinstance(f, str) else f[1]
+                    )
+                    for f in rstobject["fields"]
+                }
+
+                self.state.append(
+                    n.Field((line,), [], field_name, supported_fields[field_name])
+                )
+                return
+            except AssertionError:
```

Comment:
Assertions shouldn't really be used this way: assert statements MAY be turned off at runtime (see: https://docs.python.org/3/using/cmdline.html#cmdoption-o), so they must only be used for truly invalid states where the only safe thing to do is abort.

Can you just use an if statement for this?